packages/shared_preferences/shared_preferences_windows/lib/shared_preferences_windows.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show json;
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+
+import 'src/win32.dart' as win32;
+
+/// Wrapper to the win32 ffi functions for testing.
+class Win32Wrapper {
+  /// Calls the win32 function for [getLocalDataPath].
+  String getLocalDataPath() {
+    return win32.getLocalDataPath();
+  }
+
+  /// Calls the win32 function for [getModuleFileName].
+  String getModuleFileName() {
+    return win32.getModuleFileName();
+  }
+}
+
+/// The Windows implementation of [SharedPreferencesStorePlatform].
+///
+/// This class implements the `package:shared_preferences` functionality for the web.
+class SharedPreferencesWindows extends SharedPreferencesStorePlatform {
+  /// The name of the parent directory for [fileName].
+  final String fileDirectory = 'Flutter';
+
+  /// File system used to store to disk. Exposed for testing only.
+  @visibleForTesting
+  FileSystem fs = LocalFileSystem();
+
+  /// Wrapper for win32 ffi calls.
+  @visibleForTesting
+  Win32Wrapper win32Wrapper = Win32Wrapper();
+
+  String _localDataFilePath;
+  /// The path to the file in disk were the preferences are stored.
+  @visibleForTesting
+  String get getLocalDataFilePath {
+    if (_localDataFilePath != null) {
+      return _localDataFilePath;
+    }
+    _localDataFilePath = win32Wrapper.getLocalDataPath();
+    String appName = _getAppName();
+    // Append app-specific identifiers.
+    _localDataFilePath += '\\$fileDirectory\\$appName.ini';
```

Comment:
should use a json file extension for json data, not an ini file extension, which has its own data format: https://en.wikipedia.org/wiki/INI_file